### PR TITLE
Improve precision of mathematical operations

### PR DESCRIPTION
**Pull request by simonvbrae**
Used the library decimal.js to do mathematical operations for us.

Notes:
* Option to avoid Decimal operation has not been implemented yet.
* I may have missed some operations.
This PR changes the following implementations from the RegularFunctions file: multiplication, division, addition, substraction.
* Could benefit from a more efficient implementation. Maybe using Decimals internally?

Fixes SUM test from https://github.com/comunica/comunica/issues/287

### CODE REVIEW

File: lib/functions/RegularFunctions.ts [hunk 1]
```diff
@@ -52,36 +53,41 @@ const unaryMinus = {
 const multiplication = {
   arity: 2,
   overloads: declare()
-    .arithmetic((left, right) => left * right)
+  // @ts-ignore
```
Comment (rubensworks) [inline lib/functions/RegularFunctions.ts]: Are these necessary?

**Comment by simonvbrae** [comment]
Notes:
* I may have missed some operations. This PR only changes the following functions from the RegularFunctions file: multiplication, division, addition, substraction.
* Typescript typings aren't added yet, hence the `@ts-ignore`'s
* I think this can be implemented more efficiently by rewriting it so the library uses Decimal instances internally, instead of number

File: lib/functions/RegularFunctions.ts (hunk 1 above)
Comment (simonvbrae) [reply lib/functions/RegularFunctions.ts]: I haven't added TS typings yet but I'm going to look into it now!

Comment (simonvbrae) [reply lib/functions/RegularFunctions.ts]: TS gives the error message that .mul, or any other function, is not defined in the module.

Comment (rubensworks) [reply lib/functions/RegularFunctions.ts]: That's weird. Is it possible that you're using the typings incorrectly? Perhaps try importing like this: `import {mul} from 'decimal.js';`

**Comment by rubensworks** [comment]
Created #44 instead, as I don't have push access to @simonvbrae's fork.

**Pull request closed without merging**